src/unique_combinations.rs

```diff
@@ -0,0 +1,112 @@
+use std::fmt;
+
+/// An iterator to iterate through all the `n`-length combinations in an iterator.
+///
+/// See [`.unique_combinations()`](../trait.Itertools.html#method.unique_combinations) for moref information.
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
+pub struct UniqueCombinations<I: Iterator> {
+    len: usize,
+    position: Vec<usize>,
+    pool: Vec<I::Item>,
+    first: bool,
+    next_none: bool,
```

Comment:
We sometimes have members with similar semantics. They are usually called `done`. Should we call this one `done`, as well?